core/src/main/scala/org/apache/spark/scheduler/TaskResultGetter.scala

```diff
@@ -135,8 +135,9 @@ private[spark] class TaskResultGetter(sparkEnv: SparkEnv, scheduler: TaskSchedul
               logError(
                 "Could not deserialize TaskEndReason: ClassNotFound with classloader " + loader)
             case ex: Exception => // No-op
+          } finally {
+            scheduler.handleFailedTask(taskSetManager, tid, taskState, reason)
```

Comment:
Can you add a comment here saying something like:
"If there's an error while deserializing the TaskEndReason, this Runnable will die.  Still tell the scheduler about the task failure, to avoid a hang where the scheduler thinks the task is still running."